test/e2e/operator_test.go

```diff
@@ -226,6 +212,101 @@ func TestAWSLoadBalancerControllerWithDefaultIngressClass(t *testing.T) {
+	t.Log("Creating aws load balancer controller instance with credentials secret")
+
+	// provision AWS credentials for ALBC imitating the manually created secret
+	albcSecretName := "manual-aws-credentials-e2e"
+	albcSecretRef := corev1.ObjectReference{
+		Name:      albcSecretName,
+		Namespace: "aws-load-balancer-operator",
+	}
+	albcCrName := types.NamespacedName{
+		Name:      "mimic-manual-creds",
+		Namespace: "openshift-cloud-credential-operator",
+	}
+	albcCr, err := albcCredentialsRequest(albcCrName, albcSecretRef, "testsa")
+	if err != nil {
+		t.Fatalf("failed to generate desired credentials request to mimic manually created albc credentials: %v", err)
+	}
+	if err := kubeClient.Create(context.TODO(), albcCr); err != nil {
+		t.Fatalf("failed to create credentials request to mimic manually created albc credentials: %v", err)
+	}
+	defer func() {
+		waitForDeletion(t, kubeClient, albcCr, defaultTimeout)
+	}()
+
+	name := types.NamespacedName{Name: "cluster", Namespace: "aws-load-balancer-operator"}
+	alb := newALBCBuilder().withName(name).withCredSecret(albcSecretName).build()
+	if err := kubeClient.Create(context.TODO(), alb); err != nil && !errors.IsAlreadyExists(err) {
+		t.Fatalf("failed to create aws load balancer controller %q: %v", name, err)
+	}
+	defer func() {
+		waitForDeletion(t, kubeClient, alb, defaultTimeout)
+	}()
+
+	expected := []appsv1.DeploymentCondition{
+		{Type: appsv1.DeploymentAvailable, Status: corev1.ConditionTrue},
+	}
+	deploymentName := types.NamespacedName{Name: "aws-load-balancer-controller-cluster", Namespace: "aws-load-balancer-operator"}
+	if err := waitForDeploymentStatusCondition(t, kubeClient, defaultTimeout, deploymentName, expected...); err != nil {
+		t.Fatalf("did not get expected available condition for deployment: %v", err)
+	}
+
+	// verify that the secret used by ALBC is different from the default provisioned by the operator
+	if verified, err := verifyConsumedCredentialsSecret(kubeClient, deploymentName, albcSecretName); err != nil {
+		t.Fatalf("failed to verify consumed credentials secret for controller deployment: %v", err)
+	} else if !verified {
+		t.Fatalf("expected credentials secret %q is not consumed by the controller deployment %q", albcSecretName, deploymentName)
+	}
+
+	testWorkloadNamespace := "aws-load-balancer-test-default-ing"
+	echoSvc := createTestWorkload(t, testWorkloadNamespace)
+
+	t.Log("Creating Ingress Resource with default ingress class")
+	ingName := types.NamespacedName{Name: "echoserver", Namespace: testWorkloadNamespace}
+	ingAnnotations := map[string]string{
+		"alb.ingress.kubernetes.io/scheme":      "internet-facing",
+		"alb.ingress.kubernetes.io/target-type": "instance",
+	}
+	echoIng := buildEchoIngress(ingName, "alb", ingAnnotations, echoSvc)
+	err = retry.OnError(defaultRetryPolicy,
+		func(err error) bool {
+			t.Logf("retrying creation of echo ingress due to %v", err)
+			return !errors.IsAlreadyExists(err)
```

Comment:
```suggestion
			if errors.IsAlreadyExists(err) {
				return false
			}
			t.Logf("retrying creation of echo ingress due to error: %v", err)
			return true
```

Reply:
Changed here and in the other test cases.